blockchain_connector/avalon_blockchain_connector/ethereum/ethereum_connector.py

```diff
@@ -104,7 +105,13 @@ def _lookup_workers_in_kv_storage(self, worker_registry):
             if worker_lookup_result["result"]["totalCount"] != 0:
                 return worker_lookup_result["result"]["ids"]
             else:
-                logging.error("No workers found")
+                logging.error("No workers found in kv storage")
+                # Make recursive calls upto 3 times if no workers are found 
+                if attempt_counter < 3:
+                    # Sleep for 2 seconds before next call
+                    time.sleep(WORKER_SYNCUP_SLEEP_DURATION)
+                    self._lookup_workers_in_kv_storage(worker_registry,
+                                                       attempt_counter+1)
```

Comment:
Is this tested with recursive calls. IMO we should assign the return value to a variable and return this at end of this function. Else you can call this function from caller in a loop in a non-recursive fashion.
```
if attempt_counter < 3:
    # Sleep for 2 seconds before next call
    time.sleep(WORKER_SYNCUP_SLEEP_DURATION)
    worker_list = self._lookup_workers_in_kv_storage(worker_registry,
                                                       attempt_counter+1)
else 
     logging.error("Failed to lookup worker")
     return []
return worker_list
```

Reply:
Good suggestion @manojgop . Recursion is not something easy to follow at times. Though this would not be required anymore.
@manojgop @Ram-srini @manju956 Dependency of enclave-manager on lmdb was what went missing. Revisiting the startup logs helped. Was not consistently  reproducable which is why could not verify earlier.